R/utils-shared.R

```diff
@@ -190,21 +190,23 @@ testCallouts <- function(plot_obj, xlimits){
   
   testCalloutsByView <- function(plot_obj, callouts_index, view_num, xlimits_real, width_char, xrange){
     for(i in callouts_index){
-      callout_args <- plot_obj[[view_num]][[i]]
-      text_len <- nchar(callout_args$labels)
-      
-      len <- ifelse(is.null(callout_args$length), 0.1, callout_args$length)
-      
-      xend <- len * xrange * cos(2*pi*(30/360))
-      xnew <- callout_args$x + xend + (width_char * text_len) 
-      tooLong <- xnew > xlimits_real[2]
-      
-      if(any(tooLong)){
-        out <- which(tooLong)
-        notout <- which(!tooLong)
-        plot_obj[[view_num]][[i]]$angle[notout] <- NA
-        plot_obj[[view_num]][[i]]$angle[out] <- 150
-      }
+        callout_args <- plot_obj[[view_num]][[i]]
+        if (!is.na(xtfrm(callout_args$x[i]) | !is.na(xtfrm(callout_args$y[i])))) {  
```

Comment:
So with the ampersand, the logic comes out TRUE. with the pipe it comes up FALSE. Hmm.